### PR TITLE
Add square distance functions to xs.inc

### PR DESCRIPTION
**Pull request by IgnacioFDM**
And remove whitespace

### CODE REVIEW

**Comment by IgnacioFDM** [comment]
I haven't benchmarked this in pawn, but in C++ this kind of optimization usually leads to 2x speedup.

If you are trying to compare who is closest to some point, it just doesn't make sense to use distance instead of squared distance, since you only care whether the number is bigger or smaller.

**Comment by IgnacioFDM** [comment]
But to be honest I think what's needed the most is to copy all xs_ stuff to C++.

Stuff like get_distance_f is like 5x faster than xs_vec_distance. Anything to do with arrays is so slow in pawn. 

Currently you have to choose between a quite tidy and extensive library under xs.inc, or the very few and messy (as in function name and parameters) vector.inc functions.

I'd say to copy everything math related from xs.c onto amxmodx, with the math_ prefix.

**Comment by WPMGPRoSToTeMa** [comment]
I'm not sure that porting it to C++ will gain performance, because calling natives has some cost.

**Comment by WPMGPRoSToTeMa** [comment]
> If you are trying to compare who is closest to some point, it just doesn't make sense to use distance instead of squared distance, since you only care whether the number is bigger or smaller.

For this case we can introduce new native function, so it would be only 1 native call instead of 2 and no sqrts.

**Comment by IgnacioFDM** [comment]
>  so it would be only 1 native call instead of 2 

I don't see what you mean by 1 native call instead of 2, but anyway that's besides the point. That's just solving for a specific use case. This PR adds a basic vector function that is available in pretty much any game engine

File: plugins/include/xs.inc [hunk 1]
```diff
@@ -627,13 +627,44 @@ stock Float:xs_vec_distance_2d(const Float:vec1[], const Float:vec2[])
 	               (vec1[1]-vec2[1]) * (vec1[1]-vec2[1]));
 }
 
+/**
+ * Computes the square of the distance between two vectors (points).
+ * This is faster than the distance.
```
Comment (Arkshine) [inline plugins/include/xs.inc]: What do you mean by that?

**Comment by Arkshine** [comment]
Well, I guess it's fine.

**Pull request merged**